src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Signed.D.cs

```diff
@@ -336,107 +336,121 @@ private static bool TryParseInt32D(ReadOnlySpan<byte> source, out int value, out
 
         private static bool TryParseInt64D(ReadOnlySpan<byte> source, out long value, out int bytesConsumed)
         {
-            if (source.Length < 1)
-            {
-                bytesConsumed = 0;
-                value = default;
-                return false;
-            }
+            long sign = 0; // 0 if the value is positive, -1 if the value is negative
+            int idx = 0;
 
-            int indexOfFirstDigit = 0;
-            int sign = 1;
-            if (source[0] == '-')
+            // We use 'nuint' for the firstChar and nextChar data types in this method because
+            // it gives us a free early zero-extension to 64 bits when running on a 64-bit platform.
+
+            nuint firstChar;
+            while (true)
             {
-                indexOfFirstDigit = 1;
-                sign = -1;
+                if ((uint)idx >= (uint)source.Length) { goto FalseExit; }
+                firstChar = (uint)source[idx] - '0';
+                if ((uint)firstChar <= 9) { break; }
 
-                if (source.Length <= indexOfFirstDigit)
+                // We saw something that wasn't a digit. If it's a '+' or a '-',
+                // we'll set the 'sign' value appropriately and resume the "read
+                // first char" loop from the next index. If this loops more than
+                // once (idx != 0), it means we saw a sign character followed by
+                // a non-digit character, which should be considered an error.
+
+                if (idx != 0)
                 {
-                    bytesConsumed = 0;
-                    value = default;
-                    return false;
+                    goto FalseExit;
                 }
-            }
-            else if (source[0] == '+')
-            {
-                indexOfFirstDigit = 1;
 
-                if (source.Length <= indexOfFirstDigit)
+                idx++;
+
+                if ((uint)firstChar == unchecked((uint)('-' - '0')))
+                {
+                    sign--; // set to -1
```

Comment:
```suggestion
                    sign = -1;
```
Make it explicit?
The JIT will emit code like `mov rax, 0xffffffffffffffff` anyway.

Reply:
So I'd keep it as is. Thanks for the info.